component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -876,8 +878,7 @@ private boolean checkWithBackUpCodes(AuthenticationContext context, String userT
                 context.setProperty(SMSOTPConstants.CODE_MISMATCH, true);
             }
         } catch (UserStoreException e) {
-            throw new AuthenticationFailedException("Cannot find the user claim for OTP list for user : " +
-                    authenticatedUser, e);
+            log.error("Cannot find the user claim for OTP list for user : " + authenticatedUser, e);
```

Comment:
do we know whether the ex `UserStoreException` throws only when the claim is not available in the user profile?

Reply:
In this specific flow, it is the only case.

Reply:
if that the case can we move the try/catch wrapping the particular statement.

Reply:
IMHO I think we should keep this as it is since we need to handle the same `claim not available` scenario (even though it shouldn't throw the exception from there ideally) in [line 871](https://github.com/wso2-extensions/identity-outbound-auth-sms-otp/blob/ee3262d078418a6d072e0f5b3b898a18dca4a2ee/component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java#L871). WDYT?

Reply:
is there any other cases that `setUserClaimValue` would throw the same user store exception?

Reply:
Yes, there are. Those will not be triggered in this flow **at this point** because the code has to go through those cases while enabling the backup code claim and hence the only scenario is `claim not being found`.